utils/security/client.go

```diff
@@ -0,0 +1,216 @@
+
+	// authenticate if needed
+	if accesstoken == "" {
+		authToken, err := SecAuthenticate(c, KeycloakMasterRealm, KeycloakAdminClientID)
+		if err != nil || authToken == nil {
+			return err
+		}
+		accesstoken = authToken.AccessToken
+	}
+
+	// build REST request
+	url := hostname + "/auth/admin/realms/" + realm + "/clients"
+
+	// build the payload (JSON)
+	type PayloadClient struct {
+		DirectAccessGrantsEnabled bool      `json:"directAccessGrantsEnabled"`
+		PublicClient              bool      `json:"publicClient"`
+		ClientID                  string    `json:"clientId"`
+		Name                      string    `json:"name"`
+		RedirectUris              [1]string `json:"redirectUris"`
+	}
+	tempClient := &PayloadClient{
+		DirectAccessGrantsEnabled: true,
+		PublicClient:              true,
+		ClientID:                  clientid,
+		Name:                      clientid,
+	}
+
+	tempClient.RedirectUris = [...]string{redirectURL}
+	jsonClient, err := json.Marshal(tempClient)
+	payload := strings.NewReader(string(jsonClient))
+	req, err := http.NewRequest("POST", url, payload)
+
+	if err != nil {
+		return &SecError{errOpConnection, err, err.Error()}
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Cache-Control", "no-cache")
+	req.Header.Add("cache-control", "no-cache")
+	req.Header.Add("Authorization", "Bearer "+accesstoken)
+
+	// send request
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return &SecError{errOpConnection, err, err.Error()}
+	}
+	defer res.Body.Close()
+	body, _ := ioutil.ReadAll(res.Body)
+	if string(body) != "" {
+		keycloakAPIError := parseKeycloakError(string(body), res.StatusCode)
+		keycloakAPIError.Error = errOpResponseFormat
+		kcError := errors.New(string(keycloakAPIError.ErrorDescription))
+		return &SecError{keycloakAPIError.Error, kcError, kcError.Error()}
+	}
+	return nil
+}
+
+// SecClientGet : Retrieve Client information
+func SecClientGet(c *cli.Context) (*RegisteredClient, *SecError) {
+	if c.GlobalBool("insecure") {
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	}
+	hostname := strings.TrimSpace(strings.ToLower(c.String("host")))
+	realm := strings.TrimSpace(c.String("realm"))
+	accesstoken := strings.TrimSpace(c.String("accesstoken"))
+	clientid := strings.TrimSpace(c.String("clientid"))
+
+	// authenticate if needed
+	if accesstoken == "" {
+		authToken, err := SecAuthenticate(c, KeycloakMasterRealm, KeycloakAdminClientID)
+		if err != nil || authToken == nil {
+			return nil, err
+		}
+		accesstoken = authToken.AccessToken
+	}
+
+	// build REST request
+	url := hostname + "/auth/admin/realms/" + realm + "/clients?clientId=" + clientid
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, &SecError{errOpConnection, err, err.Error()}
+	}
+	req.Header.Add("Authorization", "Bearer "+accesstoken)
+	req.Header.Add("cache-control", "no-cache")
+	req.Header.Add("cache-control", "no-cache")
```

Comment:
The `cache-control` header is added twice. Above you add it in upper and lower case. Did you mean to do that here?

Reply:
thanks - resolved